diff --git a/src/main/index.ts b/src/main/index.ts
index 9e9244f..e08a797 100644
--- a/src/main/index.ts
+++ b/src/main/index.ts
@@ -7,7 +7,6 @@
 import "./ipc";
 
 import { app, BrowserWindow, nativeTheme } from "electron";
-import { autoUpdater } from "electron-updater";
 
 import { DATA_DIR } from "./constants";
 import { createFirstLaunchTour } from "./firstLaunch";
@@ -19,8 +18,6 @@ import { isDeckGameMode } from "./utils/steamOS";
 
 if (IS_DEV) {
     require("source-map-support").install();
-} else {
-    autoUpdater.checkForUpdatesAndNotify();
 }
 
 console.log("Vesktop v" + app.getVersion());
